### PR TITLE
fix(build): migrate setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by padamstx**
This commit migrates the project from the setup.py style of publishing to using pyproject.toml instead.

### CODE REVIEW

**Review by pyrooka — APPROVED** [review]
Everything looks good, I like this new approach of dependency management and publishing.

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,7 @@ stages:
     if: (tag IS present) AND (fork = false)
 
 # Default "install" and "script" steps.
-install:
-  - pip install setuptools=="60.8.2"
+install: true
```
[the remark this answers is not in the document]
Comment (padamstx) [reply .travis.yml:13]: For some languages, there is a default "install" step (e.g. if you don't define your own "install" step for a build job of a node.js project, then it automatically does `npm ci` for you.  So, just to be safe, if I literally don't want anything to be done during the "install" step, I typically define it simply as `install: true`

[the remark this answers is not in the document]
Comment (padamstx) [reply .travis.yml:13]: no problem... I have studied the Travis build config reference a LOT, believe it or not :joy:

**Pull request merged**